database/transaction_test.go

```diff
@@ -337,3 +359,191 @@ func testTransactionAppendToStoreAndRetrieveFromStore(t *testing.T, db database.
 			"returned wrong error: %s", testName, err)
 	}
 }
+
+func TestTransactionCommit(t *testing.T) {
+	testForAllDatabaseTypes(t, "TestTransactionCommit", testTransactionCommit)
+}
+
+func testTransactionCommit(t *testing.T, db database.Database, testName string) {
+	// Begin a new transaction
+	dbTx, err := db.Begin()
+	if err != nil {
+		t.Fatalf("%s: Begin "+
+			"unexpectedly failed: %s", testName, err)
+	}
+	defer func() {
+		err := dbTx.RollbackUnlessClosed()
+		if err != nil {
+			t.Fatalf("%s: RollbackUnlessClosed "+
+				"unexpectedly failed: %s", testName, err)
+		}
+	}()
+
+	// Put a value into the transaction
+	key := database.MakeBucket().Key([]byte("key"))
+	value := []byte("value")
+	err = dbTx.Put(key, value)
+	if err != nil {
+		t.Fatalf("%s: Put "+
+			"unexpectedly failed: %s", testName, err)
+	}
+
```

Comment:
Please also do db.Get("key") before commit, to make absolutely sure it wasn't applied before calling Commit